docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md

```diff
@@ -0,0 +1,572 @@
+        # Validate and deserialize input
+        data, errors = comment_schema.load(json_data)
+        if errors:
+            return {"status": "error", "data": errors}, 422
+        category_id = Category.query.filter_by(id=data['category_id']).first()
+        if not category_id:
+            return {'status': 'error', 'message': 'comment category not found'}, 400
+        comment = Comment(
+            category_id=data['category_id'], 
+            comment=data['comment']
+            )
+        db.session.add(comment)
+        db.session.commit()
+
+        result = comment_schema.dump(comment).data
+
+        return {'status': "success", 'data': result}, 201
+```
+ 
+Also,  Create a new file `resources/Category.py`. This will take care of all Category related Logic.
+
+```python
+from flask import request
+from flask_restful import Resource
+from Model import db, Category, CategorySchema
+
+categories_schema = CategorySchema(many=True)
+category_schema = CategorySchema()
+
+class CategoryResource(Resource):
+    def get(self):
+        categories = Category.query.all()
+        categories = categories_schema.dump(categories).data
+        return {'status': 'success', 'data': categories}, 200
+
+    def post(self):
+        json_data = request.get_json(force=True)
+        if not json_data:
+               return {'message': 'No input data provided'}, 400
+        # Validate and deserialize input
+        data, errors = category_schema.load(json_data)
+        pprint(errors)
+        if errors:
+            return errors, 422
+        category = Category.query.filter_by(name=data['name']).first()
+        if category:
+            return {'message': 'Category already exists'}, 400
+        category = Category(
+            name=json_data['name']
+            )
+
+        db.session.add(category)
+        db.session.commit()
+
+        result = category_schema.dump(category).data
+
+        return { "status": 'success', 'data': result}
+```
+
+
+Finaly, Import these resources to `app.py`. Update `app.py` exactly as below:
+
+```python
+from flask import Blueprint
+from flask_restful import Api
+from resources.Hello import Hello
+from resources.Comment import CommentResource
+from resources.Category import CategoryResource
+
+
+api_bp = Blueprint('api', __name__)
+api = Api(api_bp)
+
+# Routes
+
+api.add_resource(Hello, '/Hello')
+api.add_resource(CategoryResource, '/Category')
+api.add_resource(CommentResource, '/Comment')
+```
+
+## Step 6: Test Enpoints
+
+We now have the following endpoints available:
+
+ * POST - http://127.0.0.1:5000/api/category - for adding categories.
+
+![Creating Cetegories](../../assets/how_to_build_restful_apis_using_flask/how_to_build_restful_apis_using_flask_creating_categories.png)
+
+ * GET  -  http://127.0.0.1:5000/api/category - for listing categories.
+
+![Listing Cetegories](../../assets/how_to_build_restful_apis_using_flask/how_to_build_restful_apis_using_flask_listing_categories.png)
```

Comment:
Can't use profanity in a category

Reply:
You still haven't addressed the issue. `how_to_build_restful_apis_using_flask_listing_categories.png` is not acceptable.